refactor: Cleanup / Tidying + missing tests

NickHackman opened the pull request:
This PR is all cleanup of either confusing (spec-query.lua), unused methods/functions/files. It also moves `tests` to the top level as that's what other projects do.

## Changes

- Remove unused files
- Remove unused functions/methods
- Move `tests` to the top level (my bad, didn't know where this was supposed to go when I started)
- Don't gitignore the Gradle wrapper (this required you to pull it down multiple times)

NickHackman commented:
Going to put this further behind #41 because that does parts of this, but also implemented GitHub actions for CI and supports running whole directories

codymikol commented:
Will need another fixup, figured I'd get the features / fixes in prior to the refactor 😅

NickHackman commented:
OK, resolved conflicts and did some more stuff.

## Additional Changes
- Moved everything out of the `src/` folder
- Moved all treesitter related code into a treesitter module
- **TESTS**, I wrote a ton of tests. There's now coverage for `filter.lua` and the new `treesitter/init.lua`
- Functions now all use the same format `function M.<func_name>()`

I'll call more specifics in comments

NickHackman commented on lua/neotest-kotlin/treesitter/init.lua:
```diff
@@ -54,17 +48,28 @@ M.get_all_matches_as_string = function(path, query)
-		return results[1]
-	end
+---List all classes in a provided file using treesitter
+---@param file string
+---@return string[] classes
+function M.list_all_classes(file)
+	return get_all_matches_as_string(file, class_query)
+end
+
+---Get the first java package in a provided file using treesitter
+---@param file string
+---@return string? package
+function M.java_package(file)
+	return get_all_matches_as_string(file, package_query)[1]
+end
 
-	return nil
+---Uses neotest.treeistter.parse_positions to discover all namespaces/tests in a file.
+---@param file string
+---@return neotest.Tree?
+function M.parse_positions(file)
+	return neotest.treesitter.parse_positions(file, kotest_query, {
+		nested_namespaces = true,
+		nested_tests = false,
+	})
```
Instead of exposing the raw Treesitter queries using the `init.lua` here to only expose functions that use the treesitter queries. Moved the kotest query and parse_positions in here too. Trying to encapsulate all treesitter logic in one spot

NickHackman commented on tests/filter_spec.lua:
More tests! Really simple functions, but might as well test them now that it's in CI

NickHackman commented:
@codymikol should be good for review now, was on the same page of getting the other PR in first. Going to update the title here to include adding more tests

codymikol commented:
Not sure why, but now I'm unable to run the test examples

![image](https://github.com/user-attachments/assets/54878c42-3e97-4b90-9f27-a034089b62f3)


I'll try on some other projects

codymikol commented:
In the projects I've usually been testing with, everything still works and I'm able to see all the expected results :heavy_check_mark:

codymikol commented:
oh, the jar isn't there. I think it was before the rebase, do we want to include that in a follow up so the example project can also be run?

codymikol commented:
oh dude, I'm running the test from my own project cwd and not yours 

![image](https://github.com/user-attachments/assets/fe5b8fb6-d147-4a3d-bc88-352ccec8214b)
